--- a/src/BUILD
+++ b/src/BUILD
@@ -71,8 +71,6 @@
             ":common",
             ":sentencepiece_cc_proto",
             ":sentencepiece_model_cc_proto",
-            "@com_github_gflags_gflags//:gflags",
-            "@com_google_glog//:glog",
             "@com_google_googletest//:gtest",
             "@com_google_absl//absl/memory",
             "@com_google_absl//absl/strings",
@@ -81,7 +79,18 @@
             "//third_party/darts_clone",
             "//src/deps:threadpool",
             "//src/deps:status",
-        ],
+        ] + select({
+            "@org_tensorflow//tensorflow:android": [
+              "@org_tensorflow//tensorflow/core:tflite_portable_logging",
+            ],
+            "@org_tensorflow//tensorflow:ios": [
+              "@org_tensorflow//tensorflow/core:tflite_portable_logging",
+            ],
+            "//conditions:default": [
+                "@com_github_gflags_gflags//:gflags",
+                "@com_google_glog//:glog",
+           ],
+        }),
 )
 
 cc_library(
--- a/src/common.h
+++ b/src/common.h
@@ -21,7 +21,14 @@
 #include <utility>
 #include <vector>
 
+#if defined(ANDROID) || defined(__ANDROID__)
+#include "tensorflow/core/platform/logging.h"
+#elif TARGET_IPHONE_SIMULATOR || TARGET_OS_IPHONE
+#include "tensorflow/core/platform/logging.h"
+#else
 #include <glog/logging.h>
+#endif
+
 #include "src/deps/basic_types.h"
 #include "src/deps/canonical_errors.h"
 #include "src/deps/status.h"