diff --git a/tensorflow/core/common_runtime/BUILD b/tensorflow/core/common_runtime/BUILD
--- a/tensorflow/core/common_runtime/BUILD
+++ b/tensorflow/core/common_runtime/BUILD
@@ -235,8 +235,3 @@
         "@eigen_archive//:eigen3",
-    ] + if_static([
-        ":graph_constructor",
-        ":graph_def_builder_util",
-        "@com_google_absl//absl/algorithm:container",
-        "@com_google_absl//absl/strings",
-    ]),
+    ],
 )
@@ -448,4 +443,4 @@
     name = "copy_tensor",
-    srcs = ["copy_tensor.cc"],
+    srcs = [],
     hdrs = ["copy_tensor.h"],
     copts = tf_copts(),
@@ -1070,6 +1065,3 @@
         "@xla//xla/tsl/platform:types",
-    ] + if_static([
-        ":executor",
-        ":rendezvous_mgr",
-    ]),
+    ],
 )
@@ -2113,5 +2105,5 @@
         "//tensorflow/core/grappler:grappler_item",
         "@com_google_absl//absl/synchronization",
-    ] + if_static([":core_cpu_impl"]) + tf_protos_all() + tf_protos_grappler(),
+    ] + tf_protos_all() + tf_protos_grappler(),
 )
@@ -2130,5 +2122,3 @@
     name = "core_cpu_internal",
-    srcs = [
-        "graph_execution_state.cc",
-    ],
+    srcs = [],
     hdrs = [
@@ -2152,10 +2142,4 @@
         "@com_google_absl//absl/synchronization",
         "@eigen_archive//:eigen3",
-    ] + tf_additional_core_deps() + if_static([
-        ":core_cpu_impl",
-        "//tensorflow/core:function_ops_op_lib",
-        "//tensorflow/core:functional_grad",
-        "//tensorflow/core:functional_ops_op_lib",
-        "//tensorflow/core/kernels:required",
-    ]),
+    ] + tf_additional_core_deps(),
     alwayslink = 1,
